wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -1041,6 +1042,8 @@ public void finalizeWipe() {
         // wallet must be null for the OnboardingActivity flow
         log.info("removing wallet from memory during wipe");
         wallet = null;
+        walletBalanceObserver.close();
+        walletBalanceObserver = null;
```

Comment:
when the wallet is replaced or reset, we should also close out the balance observer.